### PR TITLE
Update samples for release

### PR DESCRIPTION
**Pull request by fabianlindfors**
With this PR we update the samples to be ready for use after release. This includes updating the image name and combining the basic auth secret, deployment, and service into one resource file, which simplifies the documentation.

I'm doubtful to the name of this unified resource file. Currently it's called `deployment.yaml` which is a bit misleading, couldn't think of anything else though.

### CODE REVIEW

File: samples/kubernetes/deployment.yaml [hunk 1]
```diff
@@ -1,3 +1,16 @@
+# Secret used to set up HTTP basic auth between the service broker and the service catalog.
+apiVersion: v1
+kind: Secret
+metadata:
+  name: atlas-service-broker-auth
+type: Opaque
+stringData:
+  username: username
+  password: password
```
Comment (BenElgar) [inline samples/kubernetes/deployment.yaml]: Just for the sake of extreme clarity, make it clear that these can be any string. I'd say do something like this:
```
stringData:
  username: my-username
  password: my-password
```

File: samples/kubernetes/deployment.yaml [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+
+---
+
```
Comment (BenElgar) [inline samples/kubernetes/deployment.yaml]: Would remove this blank line—to me it implies that the dashes are a separator when they're not.

File: samples/kubernetes/deployment.yaml [hunk 3]
```diff
@@ -23,9 +36,15 @@ spec:
           - name: BROKER_HOST
```
[the remark this answers is not in the document]
Comment (fabianlindfors) [reply samples/kubernetes/deployment.yaml]: Good point! Installed a formatter and now it looks much nicer.

File: samples/kubernetes/deployment.yaml (hunk 2 above)
Comment (fabianlindfors) [reply samples/kubernetes/deployment.yaml]: Fixed by formatter

File: samples/kubernetes/deployment.yaml (hunk 1 above)
Comment (fabianlindfors) [reply samples/kubernetes/deployment.yaml]: Changed!

**Pull request merged**